src/main/java/com/emc/pravega/perf/PravegaStreamHandler.java

```diff
@@ -0,0 +1,162 @@
+    StreamConfiguration streamconfig;
+    ReaderGroupManager readerGroupManager;
+    ReaderGroup readerGroup;
+    ScheduledExecutorService bgexecutor;
+    final int segCount;
+    final int timeout;
+
+
+    PravegaStreamHandler(String scope, String stream,
+                         String uri, int segs,
+                         int timeout,  ControllerImpl contrl,
+                         ScheduledExecutorService bgexecutor) throws Exception {
+        this.scope = scope;
+        this.stream = stream;
+        this.controllerUri= uri;
+        this.controller = contrl;
+        this.segCount = segs;
+        this.timeout = timeout;
+        this.readerGroup = null;
+        this.bgexecutor = bgexecutor;
+        streamManager = StreamManager.create(new URI(uri));
+        streamManager.createScope(scope);
+        streamconfig = StreamConfiguration.builder().scope(scope).streamName(stream)
+                        .scalingPolicy(ScalingPolicy.fixed(segCount))
+                        .build();
+
+    }
+
+
+    boolean create() {
+        return streamManager.createStream(scope, stream,streamconfig);
+    }
+
+    void scale() throws InterruptedException, ExecutionException, TimeoutException {
+        StreamSegments segments = controller.getCurrentSegments(scope, stream).join();
+        final int nseg = segments.getSegments().size();
+        System.out.println("Current segments of the stream: "+stream+ " = " + nseg);
+
+        if (nseg == segCount)  return;
+
+        System.out.println("The stream: " + stream + " will be manually scaling to "+ segCount+ " segments");
+
+        /*
+         * Note that the Upgrade stream API does not change the number of segments;
+         * but it indicates with new number of segments.
+         * after calling update stream , manual scaling is required
+         */
+        if (!streamManager.updateStream(scope, stream, streamconfig)) {
+            throw new InterruptedException("Could not able to update the stream: " + stream + " try with another stream Name");
+        }
+
+        final double keyRangeChunk = 1.0 / segCount;
+        final Map<Double, Double> keyRanges = IntStream.range(0, segCount)
+                .boxed()
+                .collect(
+                        Collectors
+                                .toMap(x -> x * keyRangeChunk,
+                                        x->(x + 1) * keyRangeChunk));
+        final List<Long> segmentList = segments.getSegments()
+                .stream()
+                .map(Segment::getSegmentId)
+                .collect(Collectors.toList());
+
+        CompletableFuture<Boolean> scaleStatus = controller.scaleStream(new StreamImpl(scope,stream),
+                segmentList,
+                keyRanges,
+                bgexecutor).getFuture();
+
+
+        if (!scaleStatus.get(timeout, TimeUnit.SECONDS)){
+            throw new InterruptedException("ERROR : Scale operation on stream "+ stream+" did not complete");
```

Comment:
You should use timeoutException. Interrupted refers to a Thread.interupt which is different. (anywhere else this pattern applies too)